### DISCUSSION TITLE
Realtime updates?

### DISCUSSION THREAD

**Discussion by 9mm** [stevearc/dressing.nvim#127] [General]
```lua
-- find/replace word under cursor
vim.keymap.set('n', '<Leader>rn', function()
  local current_word = vim.fn.expand("<cword>")
  vim.ui.input({
    prompt = 'Rename ' .. current_word .. ': ',
    default = current_word,
  }, function(new_name)
    if new_name then
      local escaped_word = vim.fn.escape(current_word, '\\/|&')
      local substitute_cmd = ":%s/\\<" .. escaped_word .. "\\>/" .. new_name .. "/g"
      vim.cmd(substitute_cmd)
    end
  end)
end, { desc = 'Rename word under cursor' })
```

I was wanting to do something like this where as you type an update it updates the view in realtime (like how you can do if I used `%s`. Is this something that could be done?

**Comment by stevearc** [comment]
Ah it's possible, but only with some hacks. This plugin does something like that: https://github.com/smjonas/inc-rename.nvim

The reason there's no _good_ way to do it is that the built-in `vim.ui.input` API doesn't have any affordance for real-time updates (like an on_key or on_change callback). Because that's not part of the `vim.ui.input` interface, it's not something that dressing supports either.

**Reply by 9mm** [reply]
ahh ok, thanks!

I actually was looking for simply a normal rename not an LSP rename, because theres so many cases where i want to rename everything, not just variable names (for example LSP doesnt rename hash keys etc)

Ill see if i can figure out how to do it from his source, thanks!
